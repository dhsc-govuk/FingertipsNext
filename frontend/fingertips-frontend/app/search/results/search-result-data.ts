export interface SearchResultInterface {
  id: number;
  topic: string;
  latestDataPeriod: string;
  dataSource: string;
  lastUpdated: string;
}

export const MOCK_DATA: SearchResultInterface[] = [
  {
    id: 1,
    topic: 'NHS',
    latestDataPeriod: '2023',
    dataSource: 'NHS website',
    lastUpdated: formatDate(new Date('December 6, 2024')),
  },
  {
    id: 2,
    topic: 'DHSC',
    latestDataPeriod: '2022',
    dataSource: 'Student article',
    lastUpdated: formatDate(new Date('November 5, 2023')),
  },
];

function formatDate(date: Date): string {
  const day = String(date.getDate()).padStart(2, '0');
  const month = date.toLocaleString('en-GB', { month: 'long' });
  const year = date.getFullYear();

  return `${day} ${month} ${year}`;
}

export const getSearchData = (): SearchResultInterface[] => {
  return MOCK_DATA;
};
