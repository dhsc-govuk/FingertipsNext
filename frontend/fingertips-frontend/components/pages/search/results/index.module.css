.result {
  padding-inline-start: 0;
}

.result li {
    border-bottom: 1px solid;
    list-style: none;
    padding: 2% 5% 0;
}

.result li:first-child {
  border-top: 1px solid
}

.result li p {
  font-size: 15px;
  line-height: 10%;
}