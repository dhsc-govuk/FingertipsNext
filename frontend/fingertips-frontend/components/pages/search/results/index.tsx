'use client';

import { BackLink, H1, Paragraph } from 'govuk-react';

export function SearchResults({ indicator }: Readonly<{ indicator: string }>) {
  return (
    <>
      <BackLink href={`/search?indicator=${indicator}`} />
      <H1>Search results</H1>
      <Paragraph>{`You searched for indicator "**${indicator}**"`}</Paragraph>
    </>
  );
}
