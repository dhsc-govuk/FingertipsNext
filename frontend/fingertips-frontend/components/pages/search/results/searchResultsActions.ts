'use server';

import { redirect, RedirectType } from 'next/navigation';

export type SomeState = {
  indicators: string[];
};

export async function viewCharts(
  prevState: SomeState,
  formData: FormData
): Promise<SomeState> {
  const indicatorsSelected = formData.getAll('indicators')?.toString();

  redirect(`/chart?indicators=${indicatorsSelected}`, RedirectType.push);
}
