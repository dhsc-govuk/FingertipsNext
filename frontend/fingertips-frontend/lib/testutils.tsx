import StyledComponentsRegistry from '@/lib/registry';

export const registryWrapper = (
  component: React.ReactNode
): React.ReactNode => {
  return <StyledComponentsRegistry>{component}</StyledComponentsRegistry>;
};
