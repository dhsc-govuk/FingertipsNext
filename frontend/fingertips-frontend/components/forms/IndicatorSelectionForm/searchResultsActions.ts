'use server';

import { z } from 'zod';
import { SearchParams, SearchStateManager } from '@/lib/searchStateManager';
import { redirect, RedirectType } from 'next/navigation';

const $SearchResultFormSchema = z.object({
  searchState: z.string(),
  indicatorsSelected: z
    .string()
    .array()
    .min(1, { message: 'Please select at least one indicator' }),
});

export type State = {
  errors?: {
    indicatorsSelected?: string[];
  };
  message?: string | null;
};

export type SearchResultForm = z.infer<typeof $SearchResultFormSchema>;

export type SearchResultState<T = SearchResultForm> = State & T;

export async function submitIndicatorSelection(
  prevState: SearchResultState,
  formData: FormData
): Promise<SearchResultState> {
  const validatedFields = $SearchResultFormSchema.safeParse({
    searchState: formData.get('searchState'),
    indicatorsSelected: formData.getAll('indicator'),
  });

  if (!validatedFields.success) {
    return {
      searchState: formData.get('searchState')?.toString() ?? '',
      indicatorsSelected: formData.getAll('indicator')?.toString().split(','),
      errors: validatedFields.error.flatten().fieldErrors,
      message: 'Please select at least one indicator',
    };
  }

  const { searchState, indicatorsSelected } = validatedFields.data;
  const state = JSON.parse(searchState);

  const searchStateManager = new SearchStateManager({
    ...state,
    [SearchParams.IndicatorsSelected]: indicatorsSelected,
  });
  redirect(searchStateManager.generatePath('/chart'), RedirectType.push);
}
