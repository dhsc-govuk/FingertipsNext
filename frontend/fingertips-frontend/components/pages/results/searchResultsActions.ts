'use server';

import { z } from 'zod';
import { SearchParams, SearchStateManager } from '@/lib/searchStateManager';
import { redirect, RedirectType } from 'next/navigation';

const $SearchResultFormSchema = z.object({
  searchedIndicator: z.string().optional(),
  indicatorsSelected: z
    .string()
    .array()
    .min(1, { message: 'Please select at least one indicator' }),
});

export type State = {
  errors?: {
    indicatorsSelected?: string[];
  };
  message?: string | null;
};

export type SearchResultForm = z.infer<typeof $SearchResultFormSchema>;

export type SearchResultState<T = SearchResultForm> = State & T;

export async function viewCharts(
  prevState: SearchResultState,
  formData: FormData
): Promise<SearchResultState> {
  const validatedFields = $SearchResultFormSchema.safeParse({
    searchedIndicator: formData.get('searchedIndicator'),
    indicatorsSelected: formData.getAll('indicator'),
  });

  if (!validatedFields.success) {
    return {
      searchedIndicator: formData.get('searchedIndicator')?.toString() ?? '',
      indicatorsSelected: formData.getAll('indicator')?.toString().split(','),
      errors: validatedFields.error.flatten().fieldErrors,
      message: 'Please select at least one indicator',
    };
  }

  const { searchedIndicator, indicatorsSelected } = validatedFields.data;

  const searchState = new SearchStateManager({
    [SearchParams.SearchedIndicator]: searchedIndicator,
    [SearchParams.IndicatorsSelected]: indicatorsSelected,
  });
  redirect(searchState.generatePath('/chart'), RedirectType.push);
}
