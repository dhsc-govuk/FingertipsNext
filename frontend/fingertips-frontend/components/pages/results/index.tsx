'use client';

import { BackLink, ErrorSummary, GridCol, GridRow, H1 } from 'govuk-react';
import { useActionState } from 'react';
import {
  IndicatorSelectionState,
  submitIndicatorSelection,
} from '../../forms/IndicatorSelectionForm/indicatorSelectionActions';
import {
  SearchParams,
  SearchStateManager,
  SearchStateParams,
} from '@/lib/searchStateManager';
import { AreaFilterPane } from '@/components/organisms/AreaFilterPane';
import { IndicatorDocument } from '@/lib/search/searchTypes';
import { AreaWithRelations } from '@/generated-sources/ft-api-client';
import { IndicatorSearchForm } from '@/components/forms/IndicatorSearchForm';
import {
  IndicatorSearchFormState,
  searchIndicator,
} from '@/components/forms/IndicatorSearchForm/indicatorSearchActions';
import { IndicatorSelectionForm } from '@/components/forms/IndicatorSelectionForm';
import { AreaFilterData } from '@/components/molecules/SelectAreasFilterPanel';

type SearchResultsProps = {
  initialIndicatorSelectionState: IndicatorSelectionState;
  searchResults: IndicatorDocument[];
  areaFilterData?: AreaFilterData;
  selectedAreasData?: AreaWithRelations[];
  searchState?: SearchStateParams;
  currentDate?: Date;
};

const generateBackLinkPath = (state?: SearchStateParams) => {
  const stateManager = SearchStateManager.initialise(state);
  return stateManager.generatePath('/');
};

export function SearchResults({
  initialIndicatorSelectionState,
  searchResults,
  areaFilterData,
  selectedAreasData,
  searchState,
  currentDate,
}: Readonly<SearchResultsProps>) {
  const [indicatorSelectionState, indicatorSelectionFormAction] =
    useActionState(submitIndicatorSelection, initialIndicatorSelectionState);

  const backLinkPath = generateBackLinkPath(searchState);

  const initialIndicatorSearchFormState: IndicatorSearchFormState = {
    indicator: searchState?.[SearchParams.SearchedIndicator] ?? '',
    searchState: JSON.stringify(searchState),
  };
  const [indicatorSearchState, indicatorSearchFormAction] = useActionState(
    searchIndicator,
    initialIndicatorSearchFormState
  );

  return (
    <>
      <BackLink href={backLinkPath} data-testid="search-results-back-link" />
      <>
        {indicatorSelectionState.message && (
          <ErrorSummary
            // description={indicatorSelectionState.message}
            errors={[
              {
                targetName: `search-results-indicator-${searchResults[0].indicatorID.toString()}`,
                text: 'Select any indicators you want to view',
              },
            ]}
            data-testid="search-result-form-error-summary"
            onHandleErrorClick={(targetName: string) => {
              const indicator = document.getElementById(targetName);
              indicator?.scrollIntoView(true);
              indicator?.focus();
            }}
          />
        )}
        <H1>
          Search results for {searchState?.[SearchParams.SearchedIndicator]}
        </H1>
        <form action={indicatorSearchFormAction}>
          <IndicatorSearchForm
            key={JSON.stringify(searchState)}
            indicatorSearchFormState={indicatorSearchState}
            searchState={searchState}
          />
        </form>
        <GridRow>
          <GridCol setWidth="one-third">
            <AreaFilterPane
              key={JSON.stringify(searchState)}
              areaFilterData={areaFilterData}
              selectedAreasData={selectedAreasData}
              searchState={searchState}
            />
          </GridCol>
          <GridCol>
            <IndicatorSelectionForm
              key={JSON.stringify(searchState)}
              searchResults={searchResults}
              searchState={searchState}
              formAction={indicatorSelectionFormAction}
              currentDate={currentDate}
            />
          </GridCol>
        </GridRow>
      </>
    </>
  );
}
