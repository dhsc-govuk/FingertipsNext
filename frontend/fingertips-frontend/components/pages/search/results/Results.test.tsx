import { render, screen, waitFor } from '@testing-library/react';
import { expect } from '@jest/globals';
import { MOCK_DATA } from '@/app/search/results/search-result-data';
import { SearchResults } from '.';
import { registryWrapper } from '@/lib/testutils';
import { userEvent } from '@testing-library/user-event';
import { viewCharts } from './searchResultsActions';

jest.mock('next/navigation', () => {
  const originalModule = jest.requireActual('next/navigation');

  return {
    ...originalModule,
    usePathname: jest.fn(),
    useSearchParams: jest.fn(),
    useRouter: jest.fn().mockImplementation(() => ({
      replace: jest.fn(),
    })),
  };
});

jest.mock('./searchResultsActions', () => ({
  viewCharts: jest.fn(),
}));

describe('Search Results Suite', () => {
  const indicator = 'test';

  it('should render elements', async () => {
    render(
      registryWrapper(
        <SearchResults indicator={indicator} searchResults={[]} />
      )
    );

    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(screen.getByText(/search results/i)).toBeInTheDocument();
    expect(screen.getAllByRole('paragraph').at(0)?.textContent).toContain(
      indicator
    );
  });

  it('should render search results', () => {
    render(
      registryWrapper(
        <SearchResults indicator={indicator} searchResults={MOCK_DATA} />
      )
    );

    expect(screen.getAllByTestId('search-result')).toHaveLength(
      MOCK_DATA.length
    );
    expect(screen.queryByText(/no results found/i)).not.toBeInTheDocument();
  });

  it('should not render elements when no indicator is entered', () => {
    render(
      registryWrapper(<SearchResults indicator="" searchResults={MOCK_DATA} />)
    );

    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(screen.getByText(/no indicator entered/i)).toBeInTheDocument();
    expect(screen.queryByRole('list')).not.toBeInTheDocument();
    expect(screen.queryAllByTestId('search-result')).toHaveLength(0);
  });

  it('should render no results found', () => {
    render(
      registryWrapper(
        <SearchResults indicator={indicator} searchResults={[]} />
      )
    );

    expect(screen.queryByText(/no results found/i)).toBeInTheDocument();
    expect(screen.queryByRole('list')).not.toBeInTheDocument();
    expect(screen.queryAllByTestId('search-result')).toHaveLength(0);
  });

  it('should mark indicators selected as checked', () => {
    render(
      <SearchResults
        indicator={indicator}
        searchResults={MOCK_DATA}
        indicatorsSelected={['1']}
      />
    );

    expect(screen.getByRole('checkbox', { name: /NHS/i })).toBeChecked();
    expect(screen.getByRole('checkbox', { name: /DHSC/i })).not.toBeChecked();
  });

  it('should provide the correct state and formData to the action', async () => {
    const user = userEvent.setup();
    const expectedFormData = new FormData();
    expectedFormData.append('indicator', '1');
    expectedFormData.append('indicator', '2');

    render(
      <SearchResults
        indicator={indicator}
        searchResults={MOCK_DATA}
        indicatorsSelected={['1']}
      />
    );

    expect(screen.getByRole('checkbox', { name: /DHSC/i })).not.toBeChecked();

    await user.click(screen.getByRole('checkbox', { name: /DHSC/i }));
    await waitFor(() => {
      expect(screen.getByRole('checkbox', { name: /DHSC/i })).toBeChecked();
    });

    const viewChartsButton = screen.getByRole('button', {
      name: /View charts/i,
    });

    await user.click(viewChartsButton);
    await waitFor(() => {
      expect(viewCharts).toHaveBeenCalledWith(
        { indicators: ['1'] },
        expectedFormData
      );
    });
  });

  it('snapshot test', () => {
    const container = render(
      registryWrapper(
        <SearchResults indicator={indicator} searchResults={MOCK_DATA} />
      )
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
