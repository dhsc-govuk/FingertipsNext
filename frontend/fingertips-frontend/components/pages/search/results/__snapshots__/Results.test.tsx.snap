// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results Suite snapshot test 1`] = `
<DocumentFragment>
  .c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: inline-block;
  position: relative;
  margin-top: 15px;
  margin-bottom: 15px;
  padding-left: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
  background-color: #ffdd00;
}

.c0:link,
.c0:visited,
.c0:hover,
.c0:active,
.c0:focus {
  color: #0b0c0c;
}

.c0[href] {
  border-bottom: 1px solid #0b0c0c;
}

.c0::before {
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 50%,100% 100%,100% 0%);
  clip-path: polygon(0% 50%,100% 100%,100% 0%);
  border-width: 5px 6px 5px 0;
  border-right-color: inherit;
  content: '';
  position: absolute;
  top: -1px;
  bottom: 1px;
  left: 0;
  margin: auto;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media print {
  .c0 {
    font-family: sans-serif;
  }
}

@media print {
  .c0:link,
  .c0:visited,
  .c0:hover,
  .c0:active,
  .c0:focus {
    color: #000;
  }
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

<a
    class="c0"
    data-testid="search-results-back-link"
    href="/search?searchedIndicator=test"
  >
    Back
  </a>
  .c0 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 32px;
  line-height: 1.09375;
  display: block;
  margin-top: 0;
  margin-bottom: 30px;
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {
  .c0 {
    color: #000;
  }
}

@media print {
  .c0 {
    font-size: 32px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 48px;
    line-height: 1.0416666666666667;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 50px;
  }
}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

<h1
    class="c0"
  >
    Search results
  </h1>
  .c0 {
  margin: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 15px;
}

.c0 > p {
  margin: 0;
}

.c0 > p > code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.c0 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c0 > pre > code {
  display: inline;
  padding: 0;
  margin: 0;
  border: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 20px;
  }
}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

<div
    class="c0"
  >
    <p>
      You searched for indicator "
      <strong>
        test
      </strong>
      "
    </p>
  </div>
  .c14 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 19px;
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  width: 100%;
  margin-top: 0;
  padding: 7px 10px;
  border: 2px solid transparent;
  border-radius: 0;
  color: #ffffff;
  background-color: #00703c;
  box-shadow: 0 2px 0 #002413;
  text-align: center;
  vertical-align: top;
  cursor: pointer;
  -webkit-appearance: none;
  margin-bottom: 22px;
}

.c14:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c14:link,
.c14:visited,
.c14:active,
.c14:hover {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14::-moz-focus-inner {
  padding: 0;
  border: 0;
}

.c14:hover,
.c14:focus {
  background-color: #00572e;
}

.c14:active {
  top: 2px;
  box-shadow: none;
}

.c14::before {
  content: "";
  display: block;
  position: absolute;
  top: -2px;
  right: -2px;
  bottom: -4px;
  left: -2px;
  background: transparent;
}

.c14:active::before {
  top: -4px;
}

.c14:disabled {
  opacity: 0.5;
  background: #00703c;
}

.c14:disabled:hover {
  background-color: #00703c;
  cursor: default;
}

.c14:disabled:focus {
  outline: none;
}

.c14:disabled:active {
  top: 0;
  box-shadow: 0 2px 0 #002413;
}

.c14 svg {
  max-width: 15px;
}

.c8 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  position: relative;
  min-height: 40px;
  padding: 0 0 0 40px;
  clear: left;
  margin-bottom: 10px;
}

.c9 {
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  opacity: 0;
  cursor: pointer;
}

.c9:checked + span:after {
  opacity: 1;
}

.c9:focus + span:before {
  outline: 3px solid transparent;
  outline-offset: 3px;
  box-shadow: 0 0 0 3px #ffdd00;
}

.c9 + span {
  opacity: 1;
  pointer-events: auto;
}

.c10 {
  display: inline-block;
  cursor: pointer;
  padding: 8px 15px 5px;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
}

.c10::before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 40px;
  height: 40px;
  border: 2px solid black;
  background: transparent;
}

.c10::after {
  content: '';
  position: absolute;
  top: 11px;
  left: 9px;
  width: 18px;
  height: 7px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
  border: solid;
  border-width: 0 0 5px 5px;
  border-top-color: transparent;
  background: transparent;
  opacity: 0;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 5px;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-top: 0;
  margin-bottom: 15px;
  list-style-type: none;
  padding-left: 0;
}

.c0 .c0 {
  margin-top: 10px;
}

.c1 > .c2 {
  margin-bottom: 5px;
}

.c11 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

.c12 {
  margin: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 15px;
}

.c12 > p {
  margin: 0;
}

.c12 > p > code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.c12 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c12 > pre > code {
  display: inline;
  padding: 0;
  margin: 0;
  border: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

.c7 {
  box-sizing: border-box;
  padding-right: 15px;
  padding-left: 15px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 -15px;
}

.c4 {
  margin: 0;
  border: 0;
  border-bottom: 1px solid #bfc1c3;
}

.c13 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 0.5;
}

.c6 {
  padding-top: 15px;
  padding-bottom: 0;
  padding-left: 25px;
  padding-right: 25px;
}

@media print {

}

@media only screen and (min-width:641px) {

}

@media print {

}

@media print {

}

@media print {
  .c14 {
    font-size: 14px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c14 {
    font-size: 19px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c14 {
    width: auto;
  }
}

@media only screen and (min-width:641px) {
  .c14 svg {
    margin-left: 10px;
  }
}

@media only screen and (min-width:641px) {
  .c14 {
    margin-bottom: 32px;
  }
}

@media print {
  .c8 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    margin-bottom: 10px;
  }
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 20px;
  }
}

@media print {

}

@media print {

}

@media only screen and (min-width:641px) {

}

@media only screen and (min-width:641px) {

}

@media print {
  .c11 {
    color: #000;
  }
}

@media print {
  .c11 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c11 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c11 {
    margin-bottom: 20px;
  }
}

@media print {
  .c12 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c12 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c12 {
    margin-bottom: 20px;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-shrink: 1;
    -ms-flex-negative: 1;
    flex-shrink: 1;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media print {
  .c13 {
    font-size: 14px;
    line-height: 0.5;
  }
}

@media only screen and (min-width:641px) {
  .c13 {
    font-size: 19px;
    line-height: 0.5;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    padding-top: 15px;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    padding-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    padding-left: 40px;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    padding-right: 40px;
  }
}

<form
    action="/action"
  >
    <input
      hidden=""
      name="searchedIndicator"
      value="test"
    />
    <ul
      class="c0 c1"
    >
      <li
        class="c2 c3"
      >
        <hr
          class="c4"
        />
      </li>
      <li
        class="c2 c3"
        data-testid="search-result"
      >
        <div
          class="c5 c6"
        >
          <div
            class="c7"
          >
            <label
              class="c8"
            >
              <input
                class="c9"
                data-testid="search-results-indicator-1"
                id="search-results-indicator-1"
                name="indicator"
                type="checkbox"
                value="1"
              />
              <span
                class="c10"
              >
                <h5
                  class="c11"
                >
                  NHS
                </h5>
                <div
                  class="c12 c13"
                >
                  <p>
                    Latest data period: 2023
                  </p>
                </div>
                <div
                  class="c12 c13"
                >
                  <p>
                    Data source: NHS website
                  </p>
                </div>
                <div
                  class="c12 c13"
                >
                  <p>
                    Last updated: 06 December 2024
                  </p>
                </div>
              </span>
            </label>
          </div>
        </div>
        <hr
          class="c4"
        />
      </li>
      <li
        class="c2 c3"
        data-testid="search-result"
      >
        <div
          class="c5 c6"
        >
          <div
            class="c7"
          >
            <label
              class="c8"
            >
              <input
                class="c9"
                data-testid="search-results-indicator-2"
                id="search-results-indicator-2"
                name="indicator"
                type="checkbox"
                value="2"
              />
              <span
                class="c10"
              >
                <h5
                  class="c11"
                >
                  DHSC
                </h5>
                <div
                  class="c12 c13"
                >
                  <p>
                    Latest data period: 2022
                  </p>
                </div>
                <div
                  class="c12 c13"
                >
                  <p>
                    Data source: Student article
                  </p>
                </div>
                <div
                  class="c12 c13"
                >
                  <p>
                    Last updated: 05 November 2023
                  </p>
                </div>
              </span>
            </label>
          </div>
        </div>
        <hr
          class="c4"
        />
      </li>
    </ul>
    <button
      class="c14"
      data-testid="search-results-button-submit"
      type="submit"
    >
      View charts
    </button>
  </form>
</DocumentFragment>
`;
